File: src/PointCar.java
/** PointCar subclass */
class PointCar extends Car {

    /** PointCar constructor */
    private PointCar(Map m, int x, int y, double mss) {
        super(m, x, y, mss);
    }

    /** Modify attributes to reflect one step */
    void step(double accAction, double steerAction) {
        angle       = (angle + steerAction) % 360;
        acc         = Math.max(Math.min(accAction, maxVel - vel), -vel);
        double dist = vel + 0.5 * acc;
        double dx   = dist * Math.cos(Math.toRadians(angle));
        double dy   = dist * Math.sin(Math.toRadians(angle));
        x          += dx;
        y          += dy;
        vel         = Math.max(Math.min(vel, maxVel), 0);
    }

}

File: gym_driver/src/PointCar.py
from numpy import sin, cos, deg2rad

from Car import Car, MAX_VEL

# PointCar class
class PointCar(Car):

    # Modify attributes to reflect one game step
    def step(self, action):
        if action is None:
            action_acc, action_angle = 0.0, 0.0
        action_acc, action_angle = action
        # Makes sure angle within (0, 360)
        self.angle = (self.angle + action_angle) % 360
        # Clamps acceleration
        self.acc = max(min(action_acc, MAX_VEL - self.vel), -self.vel)
        dist = self.vel + 0.5 * self.acc
        dx = dist * cos(deg2rad(self.angle))
        dy = dist * sin(deg2rad(self.angle))
        self.x += dx
        self.y += dy
        self.vel = max(min(self.vel + self.acc, MAX_VEL), 0)
